toolchest/rpm/utils.py

```diff
@@ -131,7 +133,7 @@ def dlrn_label_compare(l, r):
     lv = l[1]
     lr = l[2]
 
-    dlrn_regex = r'^0(\.[0-9]{1,2})?\.[0-9]{14}\.'
+    dlrn_regex = r'^[012](\.[0-9]{1,2})?\.[0-9]{14}\.[0-9a-f]{7}\.'
```

Comment:
Should we redirect the call to label_compare instead of patching it here that way results are consistent when we remove this function?

Reply:
Agree that would make sense, though I am a bit confused - could we have been using rpmLabelCompare all along? Below seems to remove the call to this function and simple call rpmLabelCompare instead.

Reply:
This fixes a corner case where "0.2.1-1.dlrn_ver" was considered newer than "0.2.2-0.dlrn_ver". While I did deprecate the function, I do think it should work correctly until it's removed.

I've updated the PR and added tests for a couple previously-wrong results for dlrn_label_compare